infrastructure/src/main/java/org/corfudb/infrastructure/log/SegmentManager.java

```diff
@@ -0,0 +1,449 @@
+     * @return the stream log segment for that ordinal
+     */
+    StreamLogSegment getStreamLogSegmentByOrdinal(long ordinal) {
+        StreamLogSegment segment = streamSegmentCache.get(ordinal);
+        Objects.requireNonNull(segment).retain();
+        return segment;
+    }
+
+    /**
+     * Get the opened garbage log segment that is responsible for storing the garbage
+     * information associated with the provided address, creating a new one if the
+     * segment is not present and opened in the current cached mapping.
+     *
+     * @param address the address to open
+     * @return the garbage log segment for that address
+     */
+    GarbageLogSegment getGarbageLogSegment(long address) {
+        long ordinal = getSegmentOrdinal(address);
+        return getGarbageLogSegmentByOrdinal(ordinal);
+    }
+
+    /**
+     * Get the opened garbage log segment that is responsible for storing the
+     * data associated with the provided segment ordinal, creating a new one
+     * if the segment is not present and opened in the current cached mapping.
+     *
+     * @param ordinal the ordinal of the segment
+     * @return the garbage log segment for that ordinal
+     */
+    GarbageLogSegment getGarbageLogSegmentByOrdinal(long ordinal) {
+        GarbageLogSegment segment = garbageSegmentCache.get(ordinal);
+        Objects.requireNonNull(segment).retain();
+        return segment;
+    }
+
+    @SuppressWarnings("unchecked")
+    private <T extends AbstractLogSegment> T getLogSegment(long ordinal, Class<T> segmentType) {
+        String filePath = getSegmentFilePath(ordinal, segmentType, false);
+        CompactionMetadata metaData = getCompactionMetaData(ordinal);
+
+        AbstractLogSegment segment;
+        if (segmentType.equals(StreamLogSegment.class)) {
+            segment = new StreamLogSegment(ordinal, logParams, filePath, logSizeQuota, metaData, dataStore);
+        } else if (segmentType.equals(GarbageLogSegment.class)) {
+            segment = new GarbageLogSegment(ordinal, logParams, filePath, logSizeQuota, metaData);
+        } else {
+            throw new IllegalArgumentException("getLogSegment: Unknown segment type: " + segmentType);
+        }
+
+        // The first time we open a file we should read to the end, to load the map
+        // of entries we already have. Once the segment address space is loaded, it
+        // should be ready to accept writes.
+        segment.loadAddressSpace();
+        return (T) segment;
+    }
+
+    private CompactionMetadata getCompactionMetaData(long ordinal) {
+        return segmentCompactionMetadata.computeIfAbsent(ordinal, seg ->
+                new CompactionMetadata(ordinal));
+    }
+
+    /**
+     * Get a new compaction metadata for the segment.
+     *
+     * @param ordinal the ordinal of the segment the metadata is associated with
+     * @return the compaction metadata for the segment
+     */
+    CompactionMetadata newCompactionMetadata(long ordinal) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use explicit scoping instead of the default package private level](https://app.codacy.com/app/annymartinez.m/CorfuDB/pullRequest?prid=4651450)